hermeto/core/errors.py

```diff
@@ -104,6 +104,20 @@ class NotAGitRepo(PackageRejected):
     """A package turned out to be not a git repository."""
 
 
+class PackageWithCorruptLockfileRejected(PackageRejected):
+    """Package lock file does not match package config."""
+
+    def __init__(self, package_path: str) -> None:
+        """Initialize the error."""
+        reason = f"{package_path} contains a lock file that does not match corresponding Cargo.toml"
+        super().__init__(reason, solution=self.default_solution, docs="")
+
+    default_solution = (
+        "Consider reaching out to package maintainer to address inconsistencies between Cargo.lock"
+        " and Cargo.toml"
+    )
```

Comment:
substitute nitpicks:
`s/corresponding/the corresponding`
`s/lock file/Cargo.lock`
`s/to package maintainer/to maintainers of the dependency in question`